### PR TITLE
[5.3] The llbuild dependency should reference the 5.3 release branch

### PR DESCRIPTION
**Pull request by yonihemi**
Fixes non-CI builds of SPM 5.3

### CODE REVIEW

**Comment by yonihemi** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: Package.swift [hunk 1]
```diff
@@ -233,7 +233,7 @@ let package = Package(
 if ProcessInfo.processInfo.environment["SWIFTPM_LLBUILD_FWK"] == nil {
     if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         package.dependencies += [
-            .package(url: "https://github.com/apple/swift-llbuild.git", .branch("master")),
+            .package(url: "https://github.com/apple/swift-llbuild.git", .branch("release/5.3")),
```
Comment (tomerd) [inline Package.swift:236]: this fix seems fine to me, but we need a more systematic way of dealing with this. perhaps an env variable (or similar) that we can use across all version-locked dependencies?

Comment (yonihemi) [reply Package.swift]: How about something like this?
Keeping an explicit default is still important IMO for the 'just add a package in Xcode' case - it's reasonable for libSwiftPM users to expect that to work without extra configuration.

Using same branch for llbuild and TSC should at least eliminate the current particular issue in future, and the constant make it stand out a little as something that needs to be manually changed.
We could also shell out to git and try to guess the corresponding tag, but that can turn to an `update_checkout.py`-style beast, maybe overkill for this.

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Review by buttaface — APPROVED** [review]

**Review by abertelrud — APPROVED** [review]
Thanks for this fix!  This will also be very useful for the upcoming 5.4, so we should merge this to main (with the branch name "main" of course) and then can adjust the default to 5.4 after branching on Jan 7.

**Comment by MaxDesiatov** [comment]
@swift-ci please test macOS platform

**Comment by abertelrud** [comment]
@swift-ci please test macOS

File: Package.swift [hunk 2]
```diff
@@ -229,11 +229,13 @@ let package = Package(
 // package dependency like this but there is no other good way of expressing
 // this right now.
 
+/// When not using local dependencies, the branch to use for llbuild and TSC repositories
+let relatedDependenciesBranch = ProcessInfo.processInfo.environment["SWIFTPM_DEPS_BRANCH"] ?? "release/5.3"
```
Comment (tomerd) [inline Package.swift:233]: thanks for putting this together @yonihemi, we discussed more and think the env variable as defined here is not adding much value. lets just set the variable to the target branch string (ie `et relatedDependenciesBranch = "release/5.3"`) and open similar PR for the main branch

**Comment by yonihemi** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
thanks @yonihemi can you please do the same for `main` branch? this will help when we branch to 5.4 later in the week

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by yonihemi** [comment]
Great. Thanks @tomerd @MaxDesiatov @abertelrud @buttaface for reviews and assistance.

**Comment by tomerd** [comment]
thank you @yonihemi

**Pull request merged**